tool/tctl/common/resource_command.go

```diff
@@ -740,9 +740,23 @@ func (rc *ResourceCommand) Delete(client auth.ClientI) (err error) {
 		}
 		fmt.Printf("windows desktop service %q has been deleted\n", rc.ref.Name)
 	case types.KindWindowsDesktop:
-		if err = client.DeleteWindowsDesktop(ctx, rc.ref.Name); err != nil {
+		desktops, err := client.GetWindowsDesktopsByName(ctx, rc.ref.Name)
+		if err != nil {
 			return trace.Wrap(err)
 		}
+		deleted := false
+		for _, desktop := range desktops {
+			if desktop.GetAddr() == rc.ref.Name {
+				if err = client.DeleteWindowsDesktop(ctx, desktop.GetHostID(), rc.ref.Name); err != nil {
+					return trace.Wrap(err)
+				}
+				deleted = true
+			}
+		}
+		if !deleted {
+			return trace.NotFound("desktop %q not found", rc.ref.Name)
```

Comment:
1. This could fail for reasons other than not found, so I wouldn't assume it's a not found error.
2. I would probably return a `trace.Aggregate(...)` of all the delete errors.